pkg/rbac/sharedData.go

```diff
@@ -51,16 +55,86 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
 	Resource: "managedclusters",
 }
 
-func (cache *Cache) PopulateSharedCache(ctx context.Context) error {
+func (shared *SharedData) GetPropertyTypes() (map[string]string, error) {
+
+	// original query:
+	// select distinct key, jsonb_typeof(value) as datatype
+	// FROM search.resources,jsonb_each(data)
+	// order by datatype;
+
+	var selectDs *goqu.SelectDataset
+	//define schema:
+	schemaTable := goqu.S("search").Table("resources")
+	//data expression to get value and key
+	jsb := goqu.L("jsonb_each(?)", goqu.C("data"))
+	//select from these datasets
+	ds := goqu.From(schemaTable, jsb)
+	//select statement with orderby and distinct clause
+	selectDs = ds.Select(goqu.L("key"), goqu.L("jsonb_typeof(?)",
+		goqu.C("value")).As("datatype")).Distinct().Order(goqu.L("datatype").Asc())
+
+	query, params, err := selectDs.ToSQL()
+
+	klog.V(5).Info("Query for property datatypes: [%s] ", query)
+	rows, err := CacheInst.pool.Query(context.Background(), query, params...)
+	if err != nil {
+		klog.Errorf("Error resolving query [%s] with args [%+v]. Error: [%+v]", query, err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	var key string
+	var value string
+	resourceTypeMap := make(map[string]string)
+	for rows.Next() {
+		err = rows.Scan(&key, &value)
+
+		if err != nil {
+			klog.Errorf("Error %s retrieving rows for query:%s", err.Error(), query)
+		}
```

Comment:
If there's an error we can't process this row, so need to add `continue`.
```suggestion
		if err != nil {
			klog.Errorf("Error %s retrieving rows for query:%s", err.Error(), query)
			continue
		}
```

Reply:
Done.